File: app/src/main/java/net/wojteksz128/worktimemeasureapp/notification/NotificationIntentService.java
package net.wojteksz128.worktimemeasureapp.notification;

import android.app.IntentService;
import android.content.Intent;
import android.support.annotation.Nullable;

import net.wojteksz128.worktimemeasureapp.notification.endOfWork.EndOfWorkNotification;

public class NotificationIntentService extends IntentService {

    /**
     * Creates an IntentService.  Invoked by your subclass's constructor.
     */
    public NotificationIntentService() {
        super(NotificationIntentService.class.getName());
    }

    @Override
    protected void onHandleIntent(@Nullable Intent intent) {
        assert intent != null;
        final String action = intent.getAction();
        EndOfWorkNotification.Action.valueOf(action).doAction(this);
    }
}

File: app/src/main/kotlin/net/wojteksz128/worktimemeasureapp/notification/action/NotificationIntentService.kt
package net.wojteksz128.worktimemeasureapp.notification.action

import android.app.IntentService
import android.content.Intent

/**
 * Creates an IntentService. Invoked by your subclass's constructor.
 */
class NotificationIntentService : IntentService(NotificationIntentService::class.java.name) {

    override fun onHandleIntent(intent: Intent?) {
        val action = intent?.action
        action?.let { Action.valueOf(it)(this) }
    }
}
